src/website/app/demos/Text/examples/nested.js

```diff
@@ -0,0 +1,73 @@
+/* @flow */
+import palette from '../../../../../library/colors';
+import { createStyledComponent } from '../../../../../library/styles';
+import { mineralTheme as theme } from '../../../../../library/themes';
+import _Text from '../../../../../library/Text';
+
+const DemoLayout = createStyledComponent(
+  'div',
+  {
+    '& > *': {
+      marginBottom: '1em'
+    }
+  },
+  {
+    includeStyleReset: true
+  }
+);
+
+const Text = createStyledComponent(_Text, ({ theme }) => ({
+  backgroundColor: palette.indigo_20,
+  padding: theme.space_inset_sm
+})).withProps({ noMargins: true });
+
+export default {
+  id: 'nested',
+  title: 'Nested Components',
+  description: `Nested Text components will display with the expected styles and
+semantic elements by default. You can control this behavior with the \`inherit\`
+prop.
+
+<Callout title="Note">
+  <p key={0}>
+    For illustrative purposes, a purple background color and padding have been
+    applied to the Text components below.
+  </p>
+</Callout>`,
+  scope: { createStyledComponent, DemoLayout, palette, Text, theme },
+  source: `
+    () => {
+      const CustomDiv = createStyledComponent('div', {
+        backgroundColor: palette.sky_10,
+        color: palette.sky_80,
+        fontFamily: 'serif',
+        fontSize: '1.25rem',
+        padding: theme.space_inset_sm
+      });
+      return (
+        <DemoLayout>
+          For non-nested Text components, you can opt <em>in</em> to inheritance
```

Comment:
Since, "inheritance" is a loaded term, I'd suggest to precede it with the word, "style" to be explicit, as in "style inheritance".  (_Here and elsewhere..._)